### PR TITLE
Kafka testing and metadata fixes

### PR DESCRIPTION
**Pull request by jaysfx**
* collectd/kafka metrics include status was synced against categorizer
* fixed unclean elections rate collection
* added missing log flush mbean collection
* bunch of metric names fixed

### CODE REVIEW

File: tests/monitors/kafka/kafka_test.py [hunk 1]
```diff
@@ -1,112 +1,163 @@
...
+    random_hex,
+    container_hostname,
+)
+from tests.helpers.verify import verify
 
 pytestmark = [pytest.mark.collectd, pytest.mark.kafka, pytest.mark.monitor_with_endpoints]
+KAFKA_METADATA = Metadata.from_package("collectd/kafka")
+PRODUCER_METADATA = Metadata.from_package("collectd/kafkaproducer")
+CONSUMER_METADATA = Metadata.from_package("collectd/kafkaconsumer")
+VERSIONS = ["0.9.0.0", "0.10.0.0", "0.11.0.0", "1.0.0", "1.0.1", "1.1.1", "2.0.0"]
```
Comment (seonsfx) [inline tests/monitors/kafka/kafka_test.py]: Any basis for all these versions? 3 `1.x`s feel like redundant to me if there's no significant diffs.

Comment (jaysfx) [reply tests/monitors/kafka/kafka_test.py]: I have no idea, I just kept what was there.

Comment (jaysfx) [reply tests/monitors/kafka/kafka_test.py]: @asuresh4 any thoughts?

Comment (seonsfx) [reply tests/monitors/kafka/kafka_test.py]: Got no comment if that's the case. Thanks

Comment (asuresh4) [reply tests/monitors/kafka/kafka_test.py]: @seonsfx @jaysfx - can't remember what the rationale was when I added these tests, maybe there was none. Looking at the release notes for kafka, I think we should be ok without the test for `1.0.1`.

**Pull request merged**